Sources/Testing/Parameterization/Test.Case.ID.swift

```diff
@@ -0,0 +1,44 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+@_spi(ExperimentalParameterizedTesting)
+extension Test.Case: Identifiable {
+  /// The ID of a test case.
+  ///
+  /// Instances of this type are considered unique within the scope of a given
+  /// parameterized test function. They are not necessarily unique across two
+  /// different ``Test`` instances.
+  public struct ID: Sendable, Equatable, Hashable {
```

Comment:
Seems to me that a test case's ID should include the ID of the test, otherwise the test cases here are going to compare equal by ID:

```swift
@Test("is negative", arguments: [0, 1, 2]) func isNegative(...)
@Test("is positive", arguments: [0, 1, 2]) func isPositive(...)
```

Reply:
I made note of this in the symbol docs for this `ID` type: the scope of uniqueness is only within a test case's associated test—it's not intended to be globally unique across any arbitrary test and I don't think it needs to be. The [documentation](https://developer.apple.com/documentation/swift/identifiable) for the `Identifiable` protocol allows adopters to define their own semantics around this (**emphasis** mine):

> `Identifiable` leaves the duration and scope of the identity unspecified. Identities can have any of the following characteristics:
> - Guaranteed always unique, like UUIDs.
> - Persistently unique per environment, like database record keys.
> - Unique for the lifetime of a process, like global incrementing integers.
> - Unique for the lifetime of an object, like object identifiers.
> - **Unique within the current collection, like collection indices.**

Reply:
Right, I am aware that `Identifiable` leaves that up to the implementation, I'm just suggesting that it might actually be important to us to track the test ID here too.

Reply:
Open to doing that but not sure it's something we need currently, and it introduces a little overhead so I'd prefer to wait until we know a specific scenario which justifies it. The places where these show up are generally paired with a `Test` (or `Test.ID`) already, e.g. in events (or their snapshots).